File: backend-sge/src/main/java/com/basis/sge/dominio/Usuario.java
package com.basis.sge.dominio;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;

@Entity
@Getter
@Setter
@Table(name = "usuario")
public class Usuario implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequence_usuario")
    @SequenceGenerator(name = "sequence_usuario", sequenceName = "sq_usuario", initialValue = 1, allocationSize = 1)
    private Integer id;

    @Column(name = "nome")
    private String nome;

    @Column(name = "cpf")
    private String cpf;

    @Column(name = "email")
    private String email;

    @Column(name = "telefone")
    private String telefone;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    @Column(name = "data_nascimento")
    private LocalDate dataNascimento;

    @Column(name = "chave")
    private String chave;
}

File: backend-sge/frontend/src/app/dominios/usuario.ts
import { User } from '@nuvem/angular-base';

export class Usuario implements User{
    
    id: number;
    nome: string;
    cpf: string;
    email: string;
    telefone: string;
    dataNascimento: Date;
    tipoUsuario: string;
    roles: string[];
    name: string;

    constructor(){
        this.roles = ["u", "a"];
    }

}